### PR TITLE
BREAKING: Major overhaul

### PR DESCRIPTION
**Pull request by treydock**
* Only support EL8 and EL9
* Only support Puppet 7 and Puppet 8
* Manage GRUB2 kernel via grubby

### CODE REVIEW

File: .github/workflows/ci.yaml [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+
+jobs:
+  unit:
+    runs-on: ubuntu-latest
+    continue-on-error: ${{ matrix.allow_failure }}
+    strategy:
+      fail-fast: false
+      matrix:
+        include:
+          - ruby: 2.7.7
```
Comment (ghoneycutt) [inline .github/workflows/ci.yaml]: 2.7.8 - https://www.puppet.com/docs/puppet/7/platform_lifecycle.html

File: .github/workflows/ci.yaml [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+            puppet: 7
+            fixtures: .fixtures.yml
+            allow_failure: false
+          - ruby: 3.2.2
```
Comment (ghoneycutt) [inline .github/workflows/ci.yaml]: 3.2.3 - https://www.puppet.com/docs/puppet/8/platform_lifecycle.html

File: metadata.json [hunk 3]
```diff
@@ -10,43 +10,43 @@
   "dependencies": [
     {
       "name": "puppetlabs/stdlib",
-      "version_requirement": ">= 4.13.1 <5.0.0"
+      "version_requirement": ">= 4.13.1 <10.0.0"
```
Comment (ghoneycutt) [inline metadata.json:13]: You could use this breaking change to bump up the lowest level of stdlib you support.

File: .github/workflows/ci.yaml (hunk 1 above)
Comment (treydock) [reply .github/workflows/ci.yaml]: Fixed

File: .github/workflows/ci.yaml (hunk 2 above)
Comment (treydock) [reply .github/workflows/ci.yaml]: Fixed

File: metadata.json (hunk 3 above)
Comment (treydock) [reply metadata.json:13]: Technically 4.x still works.

**Pull request merged**